admin-api-server/src/analytics/service/analytics.service.ts

```diff
@@ -46,32 +49,72 @@ export class AnalyticsService {
   ): Promise<MetricEntity[]> {
     const qryRes = await this.#timeseries(params);
 
-    return qryRes.rows.map(
-      (row: any) =>
-        <MetricEntity>{
-          timestamp: Math.floor(row['timestamp'].getTime() / 1000),
-          value: Number(row['nft_count']),
-        },
-    );
+    const nftCountMap: Record<string, number> = {};
+    qryRes.rows.forEach((row: any) => {
+      const key = Math.floor(row['timestamp'].getTime() / 1000).toString();
+      nftCountMap[key] = Number(row['nft_count']);
+    });
+
+    return this.getTimestampData(nftCountMap, params.resolution);
   }
 
   async getTimeseriesSalesPriceVolume(
     params: MetricParams,
   ): Promise<MetricEntity[]> {
     const qryRes = await this.#timeseries(params);
 
-    return qryRes.rows.map(
-      (row: any) =>
-        <MetricEntity>{
-          timestamp: Math.floor(row['timestamp'].getTime() / 1000),
-          value: Number(
-            this.currencyService.convertToCurrency(
-              Number(row['price_volume']),
-              BASE_CURRENCY,
-            ),
-          ),
-        },
-    );
+    const priceVolumeMap: Record<string, number> = {};
+    qryRes.rows.forEach((row: any) => {
+      const key = Math.floor(row['timestamp'].getTime() / 1000).toString();
+
+      priceVolumeMap[key] = Number(
+        this.currencyService.convertToCurrency(
+          Number(row['price_volume']),
+          BASE_CURRENCY,
+        ),
+      );
+    });
+
+    return this.getTimestampData(priceVolumeMap, params.resolution);
+  }
+
+  getTimestampData(
+    timestampMap: Record<string, number>,
+    resolution: Resolution,
+  ): MetricEntity[] {
+    let manipulate: ManipulateType;
+    switch (resolution) {
+      case Resolution.Hour:
+        manipulate = Resolution.Hour as ManipulateType;
+        break;
+      case Resolution.Day:
+        manipulate = Resolution.Day as ManipulateType;
+        break;
+      case Resolution.Week:
+        manipulate = Resolution.Week as ManipulateType;
+        break;
+      case Resolution.Month:
+        manipulate = Resolution.Month as ManipulateType;
+        break;
+      default:
+        manipulate = Resolution.Day as ManipulateType;
+    }
+
+    const timestampKeys = Object.keys(timestampMap);
+    let currentDate = dayjs.unix(Number(timestampKeys.at(0))).toDate();
```

Comment:
Opinionated but perhaps nice to call these variables `...Time` instead of `...Date`. I know they're of `Date` type and therefore these chosen names but I think that this Date type's name is deceptive. Initially I read the code here as if it was tracking at which date it is, but it is tracking at which timestamp it is.

Reply:
yeah, just `current` and `end` would've also worked. i don't mind so much between that and `currenTime`, `endTime`.

Reply:
Changed it to `current`and `end`